src/components/ChattingRoom/ChatInput.tsx

```diff
@@ -0,0 +1,292 @@
+      e.preventDefault();
+      handleSubmit(e);
+    }
+  };
+
+  //줄 바뀌면 textarea 높이 같이 증가
+  const autoAdjustTextarea = (element: HTMLTextAreaElement) => {
+    element.style.height = "auto";
+    element.style.height = element.scrollHeight + "px";
+  };
+
+  // //시간 계산 관련 코드
+  const [isDateMessage, setIsDateMessage] = useState<boolean>(false);
+
+  const handleIsDate = (
+    message: { text: string; sender: number; timestamp: string },
+    index: number
+  ) => {
+    const timestamp = new Date(message.timestamp);
+    if (index === 0) {
+      setIsDateMessage(true);
+    } else if (index > 0) {
+      const prevMessage = messages[index - 1];
+      const prevTimestamp = new Date(prevMessage.timestamp);
+
+      const timeDifference = timestamp.getTime() - prevTimestamp.getTime();
+
+      if (timeDifference >= 60 * 60 * 1000) {
+        // 1시간 이상 차이
+        const hours = timestamp.getHours();
+        const minutes = timestamp.getMinutes();
+        const amOrPm = hours < 12 ? "오전" : "오후";
+        const newDate = `${amOrPm} ${hours % 12}:${minutes}`;
+        setIsDateMessage(true);
+        return newDate;
+      } else if (timeDifference >= 24 * 60 * 60 * 1000) {
+        // 1일 이상 차이
+        const month = timestamp.getMonth() + 1;
+        const day = timestamp.getDate();
+        const hours = timestamp.getHours();
+        const minutes = timestamp.getMinutes();
+        const amOrPm = hours < 12 ? "오전" : "오후";
+        const newDate = `${month}월 ${day}일 ${amOrPm} ${
+          hours % 12
+        }:${minutes}`;
+        setIsDateMessage(true);
+        return newDate;
+      } else {
+        setIsDateMessage(false);
+        return null; // 시간 차이가 1시간 미만, 1일 미만인 경우
+      }
+    }
+  };
+
+  return (
+    <>
+      <Container ref={containerRef}>
+        {messages.map((message, index) => (
+          <ChattingItem
+            key={index}
+            message={message.text}
+            sender={message.sender}
+            isLastItem={
+              index === messages.length - 1 ||
+              (index < messages.length - 1 &&
+                message.sender !== messages[index + 1].sender)
+            }
+            timestamp={message.timestamp}
+            isDateMessage={isDateMessage}
+          />
+        ))}
+      </Container>
+      <InputContainer onSubmit={handleSubmit}>
+        <InputDiv>
+          <CameraIcon src={camera} />
+          <InputBox
+            placeholder="메세지 보내기..."
+            onFocus={(e) => {
+              handleInputFocus();
+              autoAdjustTextarea(e.target);
+            }}
+            onBlur={handleInputBlur}
+            value={inputMessage}
+            onChange={(e) => {
+              setInputMessage(e.target.value);
+              autoAdjustTextarea(e.target);
+            }}
+            onKeyDown={handleKeyDown}
+            ref={inputBoxRef}
+          />
+          {isFocused ? (
+            <SendButton onClick={handleSendButtonClick}>
```

Comment:
이 부분 저도 예전에 겪었던 문제입니다! 제가 알기론 `<SendButton>` 을 누르는 순간 onBlur가 더 높은 우선순위를 가지기 때문에 onClick이 실행되기 전에 먼저 isFocused가 false가 되면서 버튼이 없어지는 거 같습니다!
onMouseDown은 onBlur보다 높은 우선순위를 가진다고 합니다!

[참고자료](https://kang-ju.tistory.com/5)
```suggestion
          {isFocused ? (
            <SendButton onMouseDown={handleSendButtonClick}>
```

Reply:
안 그래도 뭐가 문제인가 하며 이리저리 바꿔봤었는데... 🥹 좋은 방법 알려주셔서 정말 감사해요!! 활용해서 수정하도록 하겠습니다 :)